packages/data/src/components/use-select/index.js

```diff
@@ -78,9 +80,22 @@ const renderQueue = createQueue();
  *
  * @return {Function}  A custom react hook.
  */
-export default function useSelect( _mapSelect, deps ) {
-	const mapSelect = useCallback( _mapSelect, deps );
+export default function useSelect( storeKey, _mapSelect, deps ) {
+	if ( arguments.length === 3 ) {
+		[ storeKey, _mapSelect, deps ] = arguments;
+	} else {
+		// Backwards compatible signature
+		storeKey = null;
+		[ _mapSelect, deps ] = arguments;
+	}
```

Comment:
This is here to keep the arguments in README and make the linter happy. I don't like it too much. The alternative is having a new `useStoreSelect` selector and deprecate this one, but I don't like a longer name for the recommended default. Thoughts?

Reply:
Backwards compatibility code is good to have?

`useStoreSelect` would be interesting if we export it for the modules creating the store instead of importing the general `useSelect` from the data module. Tbh, I think that makes more sense than hardcoding the keys as strings. @gziolo was also working on a PR to export keys from the components creating the stores.

Reply:
It is good to have, I just don't like processing `arguments` like that.

> useStoreSelect would be interesting if we export it for the modules creating the store instead of importing the general useSelect from the data module. Tbh, I think that makes more sense than hardcoding the keys as strings. @gziolo was also working on a PR to export keys from the components creating the stores.

Sounds like same goes for `useDispatch()`. It would work, we'd just have to update all the store-registering modules and remember about exporting this in the newly created ones 🤔

Reply:
Hm..

```js
// Calls the callback with a `select` function
import { useSelect } from '@wordpress/data';

// Calls the callback with module-specific selectors:
import { useSelect as useCoreDataSelect } from '@wordpress/core-data';
import { useSelect as useEditWidgetsSelect } from '@wordpress/edit-widgets';
```

I'm not sure if I like it or not :D

Reply:
```js
// Deprecated, a function expects a module as an argument
import { useDispatch } from '@wordpress/data';

// Deprecated, a function that takes no arguments
import { useDispatch as useCoreDataDispatch } from '@wordpress/core-data';
import { useDispatch as useEditWidgetsDispatch } from '@wordpress/edit-widgets';
```

Reply:
I'm intrigued though, I'll give it a try in another PR tomorrow.